### PR TITLE
Align CMake configuration with other HSF projects

### PR DESCRIPTION
**Pull request by JavierCVilla**
- Add support for C++ Standard, considering user options
- Apply good CMake practices for compiler options
- Update podio version
- Add GNU variables
- Increase minimum CMake version to 3.3, as recommended by HSF

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,59 @@
-cmake_minimum_required(VERSION 2.8)
+# HSF recommends 3.3 to support C/C++ compile features for C/C++11 across all
+# platforms
+cmake_minimum_required(VERSION 3.3)
 
+#--- Project name --------------------------------------------------------------
 project(podio)
+
+#--- Version -------------------------------------------------------------------
 set( podio_VERSION_MAJOR 0 )
```
Comment (vvolkl) [inline CMakeLists.txt]: With cmake 3.x a nicer syntax is possible:  https://github.com/HSF/tools/blob/master/project_template/CMakeLists.txt#L7

**Comment by vvolkl** [comment]
@JavierCVilla Looks good, but I would prefer to first merge #72

**Comment by vvolkl** [comment]
Hi Javier, my changes are missing after your force-push

**Comment by JavierCVilla** [comment]
@vvolkl Fixed, thanks for the heads-up

**Comment by vvolkl** [comment]
Great, podio looks to be in good shape now. I'll prepare a tag.

**Pull request merged**